{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Creates granule images based on .csv, .ims file and fourier.h5 file "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "import h5py\n",
    "import pandas as pd\n",
    "import os\n",
    "from shapely.geometry import Polygon\n",
    "import plotly.express as px\n",
    "import plotly.graph_objects as go\n",
    "import numpy as np\n",
    "from IPython.display import Image\n",
    "\n",
    "from helper_functions.helper_functions import get_coords, pixels_between_points\n",
    "# from dataset_creation.helper_functions.helper_functions import get_coords, pixels_between_points\n",
    "from helper_functions import frame_gen as fg\n",
    "from helper_functions.frame_gen import startVM, vmManager\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib.ticker import MaxNLocator, AutoLocator\n",
    "import matplotlib.colors as mcol\n",
    "font = {'family' : 'serif',\n",
    "         'size'   : 26,\n",
    "         'serif':  'cmr10'\n",
    "         }\n",
    "plt.rc('font', **font)\n",
    "plt.rc('axes', unicode_minus=False)\n",
    "plt.rcParams.update({'font.size': 26})\n",
    "pd.options.mode.chained_assignment = None\n",
    "\n",
    "from pathlib import Path\n",
    "import pathlib\n",
    "temp = pathlib.PosixPath\n",
    "pathlib.PosixPath = pathlib.WindowsPath"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# .csv | Base file name for .ims and .h5, Frame ID, Granule ID, create plots based on these with Model prediction  \n",
    "\n",
    "def read_csv(base_filename):\n",
    "    \"Reads in csv with what granules to plot, their origin file, frame id, granule id.\"\n",
    "    base_csv = pd.read_csv(base_filename, index_col=0)\n",
    "    return base_csv.reset_index(drop=True)\n",
    "    # comp_df['file'] = pathlib.Path(full_path).stem \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_10.50.34--NControlLongB--T1015-Broken.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_10.50.34--NControlLongB--T1015-Broken.ims'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_10.59.53--NControlLongB--T1015-Broken.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_10.59.53--NControlLongB--T1015-Broken.ims'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.15.39--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.15.39--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.17.30--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.17.30--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.19.22--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.19.22--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.23.06--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.23.06--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.24.54--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.24.54--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.26.51--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.26.51--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.28.41--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.28.41--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.30.27--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.30.27--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.32.21--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.32.21--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.34.46--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.34.46--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.38.35--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.38.35--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.40.27--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.40.27--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.42.32--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.42.32--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.46.17--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.46.17--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.48.25--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.48.25--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.50.27--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.50.27--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.55.00--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.55.00--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.56.54--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.56.54--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_11.58.48--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_11.58.48--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.03.04--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.03.04--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.07.15--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.07.15--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.09.17--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.09.17--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.11.05--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.11.05--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.14.52--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.14.52--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.16.45--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.16.45--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.18.43--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.18.43--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.22.31--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.22.31--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.24.27--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.24.27--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.26.16--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.26.16--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.28.17--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.28.17--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.30.04--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.30.04--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.32.07--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.32.07--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.39.41--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.39.41--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.49.52--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.49.52--NControlLongB--T1015-Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_12.56.45--NControlLongB--T1015-Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_12.56.45--NControlLongB--T1015-Burst.ims'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.16.03--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.16.03--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.17.18--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.17.18--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.19.09--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.19.09--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.23.42--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.23.42--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.24.56--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.24.56--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.26.19--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.26.19--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.27.38--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.27.38--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.28.49--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.28.49--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.31.58--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.31.58--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.45.03--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.45.03--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.46.04--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.46.04--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.47.11--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.47.11--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.49.24--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.49.24--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.50.29--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.50.29--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.53.14--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.53.14--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.54.23--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.54.23--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.55.25--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.55.25--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_14.56.30--NCz--T1334--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_14.56.30--NCz--T1334--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.02.57--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.02.57--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.05.17--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.05.17--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.06.28--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.06.28--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.07.26--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.07.26--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.10.04--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.10.04--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.12.22--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.12.22--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.13.31--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.13.31--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.14.52--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.14.52--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.16.08--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.16.08--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.18.08--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.18.08--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.20.36--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.20.36--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.23.27--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.23.27--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.24.33--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.24.33--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.26.38--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.26.38--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.28.54--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.28.54--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.29.52--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.29.52--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.32.22--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.32.22--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.33.19--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.33.19--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.34.18--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.34.18--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.35.29--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.35.29--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.36.34--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.36.34--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.37.35--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.37.35--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.39.05--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.39.05--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.40.05--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.40.05--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.41.21--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.41.21--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.44.04--NCz--T1437--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.44.04--NCz--T1437--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.51.31--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.51.31--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.52.50--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.52.50--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.55.27--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.55.27--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_15.57.04--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_15.57.04--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_16.02.40--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_16.02.40--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_16.04.42--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_16.04.42--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_16.06.28--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_16.06.28--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_16.07.41--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_16.07.41--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_16.11.45--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_16.11.45--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_16.13.00--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_16.13.00--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_16.14.20--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_16.14.20--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_16.17.58--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_16.17.58--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-12-09_16.19.07--NCz--T1518--GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-12-09_16.19.07--NCz--T1518--GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.18.49--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.18.49--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.19.59--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.19.59--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.20.16--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.20.16--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.20.33--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.20.33--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.21.32--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.21.32--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.22.31--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.22.31--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.24.00--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.24.00--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.25.05--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.25.05--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.26.01--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.26.01--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.27.05--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.27.05--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.31.38--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.31.38--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.32.49--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.32.49--NAs--T1354-GFP_Burst.ims'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.39.43--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.39.43--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.40.44--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.40.44--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.41.42--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.41.42--NAs--T1354-GFP_Burst.ims'\n",
      "'NoneType' object has no attribute 'find_class'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.44.01--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.44.01--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.45.04--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.45.04--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.46.06--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.46.06--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.47.08--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.47.08--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.48.07--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.48.07--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.49.14--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.49.14--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.50.18--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.50.18--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.51.27--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.51.27--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.52.29--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.52.29--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.53.32--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.53.32--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.55.45--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.55.45--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.57.00--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.57.00--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.58.02--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.58.02--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_14.58.58--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_14.58.58--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.00.13--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.00.13--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.01.14--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.01.14--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.02.40--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.02.40--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.03.46--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.03.46--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.04.45--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.04.45--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.05.58--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.05.58--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.06.56--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.06.56--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.07.59--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.07.59--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.09.04--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.09.04--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.10.02--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.10.02--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.11.08--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.11.08--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.14.28--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.14.28--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.15.45--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.15.45--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.19.33--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.19.33--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.20.30--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.20.30--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.21.39--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.21.39--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.22.40--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.22.40--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.23.46--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.23.46--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.24.55--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.24.55--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.28.39--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.28.39--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.32.55--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.32.55--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.35.19--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.35.19--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.36.20--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.36.20--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.37.16--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.37.16--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.39.15--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.39.15--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.40.35--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.40.35--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.41.32--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.41.32--NAs--T1354-GFP_Burst.ims'\n",
      "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2020-02-05_15.42.34--NAs--T1354-GFP_Burst.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2020-02-05_15.42.34--NAs--T1354-GFP_Burst.ims'\n"
     ]
    }
   ],
   "source": [
    "csv = read_csv(\"D:\\Master\\MasterProject\\GE_result_analysis/test.csv\")\n",
    "\n",
    "for i, file in enumerate(csv.groupby('file')):\n",
    "    filename = file[0]\n",
    "    data = file[1]\n",
    "\n",
    "    try:\n",
    "        im_directory = \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\"\n",
    "        image_path = str(f\"{im_directory}/{filename}.ims\")\n",
    "        image_gen = fg.bioformatsGen(Path(image_path))\n",
    "        next(image_gen)\n",
    "\n",
    "        for p in ['as_02_endre', 'as_31_endre', 'cz_endre']:\n",
    "            # dataframe with analysis results from .ims file\n",
    "            try:\n",
    "                image_analysed_results_df = pd.read_hdf(Path(f\"D:\\Granule_experiment_data\\ALL_FOURIER_h5/{p}/fourier/{filename}.h5\"), mode=\"r\", key=\"fourier\")\n",
    "            except:\n",
    "                break\n",
    "    except Exception as e:\n",
    "        print(e)\n",
    "        continue\n",
    "    \n",
    "    # print(filename)\n",
    "    # print(file[1].shape)\n",
    "    print(file[1])\n",
    "\n",
    "    break\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_10.50.34--NControlLongB--T1015-Broken.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_10.50.34--NControlLongB--T1015-Broken.ims'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[26], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i, im \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28menumerate\u001b[39m(image_gen):\n\u001b[0;32m      2\u001b[0m     \u001b[38;5;28mprint\u001b[39m(i, im)\n\u001b[0;32m      3\u001b[0m     \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[1;32md:\\Master\\MasterProject\\GE_result_analysis\\helper_functions\\frame_gen.py:174\u001b[0m, in \u001b[0;36mbioformatsGen\u001b[1;34m(im_path)\u001b[0m\n\u001b[0;32m    172\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\" Load an image from a bioformats file. \"\"\"\u001b[39;00m\n\u001b[0;32m    173\u001b[0m \u001b[38;5;66;03m# Get some metadata from the OMEXML data\u001b[39;00m\n\u001b[1;32m--> 174\u001b[0m md \u001b[38;5;241m=\u001b[39m \u001b[43mbf\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_omexml_metadata\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mstr\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mim_path\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    175\u001b[0m o \u001b[38;5;241m=\u001b[39m bf\u001b[38;5;241m.\u001b[39mOMEXML(md)\n\u001b[0;32m    177\u001b[0m \u001b[38;5;66;03m# Extract the relevant terms\u001b[39;00m\n",
      "File \u001b[1;32mc:\\ProgramData\\anaconda3\\envs\\granule_explorer\\lib\\site-packages\\bioformats\\formatreader.py:1019\u001b[0m, in \u001b[0;36mget_omexml_metadata\u001b[1;34m(path, url)\u001b[0m\n\u001b[0;32m   1008\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget_omexml_metadata\u001b[39m(path\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, url\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[0;32m   1009\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m'''Read the OME metadata from a file using Bio-formats\u001b[39;00m\n\u001b[0;32m   1010\u001b[0m \n\u001b[0;32m   1011\u001b[0m \u001b[38;5;124;03m    :param path: path to the file\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   1017\u001b[0m \n\u001b[0;32m   1018\u001b[0m \u001b[38;5;124;03m    '''\u001b[39;00m\n\u001b[1;32m-> 1019\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m \u001b[43mImageReader\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpath\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpath\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mperform_init\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m \u001b[38;5;28;01mas\u001b[39;00m rdr:\n\u001b[0;32m   1020\u001b[0m         \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[0;32m   1021\u001b[0m         \u001b[38;5;66;03m# Below, \"in\" is a keyword and Rhino's parser is just a little wonky I fear.\u001b[39;00m\n\u001b[0;32m   1022\u001b[0m         \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[0;32m   1023\u001b[0m         \u001b[38;5;66;03m# It is critical that setGroupFiles be set to false, goodness knows\u001b[39;00m\n\u001b[0;32m   1024\u001b[0m         \u001b[38;5;66;03m# why, but if you don't the series count is wrong for flex files.\u001b[39;00m\n\u001b[0;32m   1025\u001b[0m         \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[0;32m   1026\u001b[0m         script \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\"\"\u001b[39m\n\u001b[0;32m   1027\u001b[0m \u001b[38;5;124m        importClass(Packages.loci.common.services.ServiceFactory,\u001b[39m\n\u001b[0;32m   1028\u001b[0m \u001b[38;5;124m                    Packages.loci.formats.services.OMEXMLService,\u001b[39m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   1039\u001b[0m \u001b[38;5;124m        xml;\u001b[39m\n\u001b[0;32m   1040\u001b[0m \u001b[38;5;124m        \u001b[39m\u001b[38;5;124m\"\"\"\u001b[39m\n\u001b[0;32m   1041\u001b[0m         xml \u001b[38;5;241m=\u001b[39m jutil\u001b[38;5;241m.\u001b[39mrun_script(script, \u001b[38;5;28mdict\u001b[39m(path\u001b[38;5;241m=\u001b[39mrdr\u001b[38;5;241m.\u001b[39mpath, reader \u001b[38;5;241m=\u001b[39m rdr\u001b[38;5;241m.\u001b[39mrdr))\n",
      "File \u001b[1;32mc:\\ProgramData\\anaconda3\\envs\\granule_explorer\\lib\\site-packages\\bioformats\\formatreader.py:619\u001b[0m, in \u001b[0;36mImageReader.__init__\u001b[1;34m(self, path, url, perform_init)\u001b[0m\n\u001b[0;32m    616\u001b[0m     filename \u001b[38;5;241m=\u001b[39m os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39msplit(path)[\u001b[38;5;241m1\u001b[39m]\n\u001b[0;32m    618\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39misfile(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpath):\n\u001b[1;32m--> 619\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mIOError\u001b[39;00m(\n\u001b[0;32m    620\u001b[0m         errno\u001b[38;5;241m.\u001b[39mENOENT,\n\u001b[0;32m    621\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mThe file, \u001b[39m\u001b[38;5;130;01m\\\"\u001b[39;00m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;130;01m\\\"\u001b[39;00m\u001b[38;5;124m, does not exist.\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m path,\n\u001b[0;32m    622\u001b[0m         path)\n\u001b[0;32m    624\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstream \u001b[38;5;241m=\u001b[39m jutil\u001b[38;5;241m.\u001b[39mmake_instance(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mloci/common/RandomAccessInputStream\u001b[39m\u001b[38;5;124m'\u001b[39m,\n\u001b[0;32m    625\u001b[0m                                   \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m(Ljava/lang/String;)V\u001b[39m\u001b[38;5;124m'\u001b[39m,\n\u001b[0;32m    626\u001b[0m                                   \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpath)\n\u001b[0;32m    628\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mrdr \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] The file, \"D:\\Granule_experiment_data\\ALL_IMS\\ALL_IMS_TOGETHER\\2019-10-31_10.50.34--NControlLongB--T1015-Broken.ims\", does not exist.: 'D:\\\\Granule_experiment_data\\\\ALL_IMS\\\\ALL_IMS_TOGETHER\\\\2019-10-31_10.50.34--NControlLongB--T1015-Broken.ims'"
     ]
    }
   ],
   "source": [
    "for i, im in enumerate(image_gen):\n",
    "    print(i, im)\n",
    "    break"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "granule_explorer",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
